src/zc/lockfile/__init__.py

```diff
@@ -83,15 +90,24 @@ def __init__(self, path):
             _lock_file(fp)
         except:
             fp.seek(1)
-            pid = fp.read().strip()[:20]
+            content = fp.read().strip()
             fp.close()
-            if not pid:
-                pid = 'UNKNOWN'
-            logger.exception("Error locking file %s; pid=%s", path, pid)
+            if content_template == '{pid}':
+                # Original exception message format when using the default
+                # lock file template
+                pid = content[:20] if content else 'UNKNOWN'
+                logger.exception("Error locking file %s; pid=%s", path, pid)
+            else:
+                # Include the first 40 characters of lock file contents for
+                # custom lock file templates
+                logger.exception('Error locking file %s; content: "%s%s"',
+                                 path, content[:40],
+                                 '...' if len(content) > 40 else '')
             raise
 
         self._fp = fp
-        fp.write(" %s\n" % os.getpid())
+        fp.write(" %s\n" % content_template.format(pid=os.getpid(),
+                                                   hostname=LazyHostName()))
```

Comment:
Ok I reverted that and made the tests be more exact in that they skip the first byte, strip any trailing whitespace (just in caseOS specific line endings might become an issue), and compare contents to expected output in the doctest.

Should I squash all the commits into a neat patch or do you prefer to keep the branch history?